zallet/src/components/json_rpc/methods.rs

```diff
@@ -17,6 +22,47 @@ pub(crate) trait Rpc {
     #[method(name = "z_listaccounts")]
     async fn list_accounts(&self) -> list_accounts::Response;
 
+    /// For the given account, derives a Unified Address in accordance with the remaining
+    /// arguments:
+    ///
+    /// - If no list of receiver types is given (or the empty list `[]`), the best and
+    ///   second-best shielded receiver types, along with the \"p2pkh\" (i.e. transparent)
+    ///   receiver type, will be used.
+    /// - If no diversifier index is given, the next unused index (that is valid for the
+    ///   list of receiver types) will be selected.
+    ///
+    /// The account must have been previously generated by a call to the `z_getnewaccount`
+    /// RPC method.
+    ///
+    /// Once a Unified Address has been derived at a specific diversifier index,
+    /// re-deriving it (via a subsequent call to `z_getaddressforaccount` with the same
+    /// account and index) will produce the same address with the same list of receiver
+    /// types. An error will be returned if a different list of receiver types is
+    /// requested.
+    #[method(name = "z_getaddressforaccount")]
+    async fn get_address_for_account(
+        &self,
+        account: JsonValue,
+        receiver_types: Option<Vec<String>>,
+        diversifier_index: Option<u128>,
+    ) -> get_address_for_account::Response;
+
+    /// Lists the addresses managed by this wallet by source, including those generated
+    /// from randomness by this wallet, Sapling addresses generated from the legacy HD
+    /// seed, imported watchonly transparent addresses, shielded addresses tracked using
+    /// imported viewing keys, and addresses derived from the wallet's mnemonic seed for
+    /// releases version 4.7.0 and above.
```

Comment:
This text is copied from `zcashd`, but will need revising once we've determined what we're actually supporting.